routersploit/core/exploit/shell.py

```diff
@@ -28,14 +27,11 @@ def shell(exploit, architecture="", method="", payloads=None, **params):
     options = []
 
     if architecture and method:
-        path = "routersploit/modules/payloads/{}/".format(architecture)
-
         # get all payloads for given architecture
-        all_payloads = [f.split(".")[0] for f in listdir(path) if isfile(join(path, f)) and f.endswith(".py") and f != "__init__.py"]
+        all_payloads = [p for p in index_modules() if "payloads.{}".format(architecture) in p]
 
-        payload_path = path.replace("/", ".")
         for p in all_payloads:
-            module = getattr(importlib.import_module("{}{}".format(payload_path, p)), 'Payload')
+            module = getattr(importlib.import_module('routersploit.modules.' + p), 'Payload')
```

Comment:
With this implementation displayed names of payloads are:
```
payloads.mipsle.reverse_tcp
payloads.mipsle.bind_tcp
```
They should be:
```
mipsle/reverse_tcp
mipsle/bind_tcp
```
Its basically `p` variable that goes as a key to `available_payloads`

Reply:
Whoops, sorry for this regression and nice catch!
Commit bc7086f should address that.